tests/src/JIT/HardwareIntrinsics/X86/Sse2/AddSaturate.cs

```diff
@@ -0,0 +1,119 @@
+                    }
+
+                    for (int i = 0; i < testsCount; i++)
+                    {
+                        (Vector128<ushort>, Vector128<ushort>, Vector128<ushort>) value = ushortTable[i];
+                        var result = Sse2.AddSaturate(value.Item1, value.Item2);
+                        ushortTable.SetOutArray(result);
+                    }
+
+                    for (int i = 0; i < testsCount; i++)
+                    {
+                        (Vector128<sbyte>, Vector128<sbyte>, Vector128<sbyte>) value = sbyteTable[i];
+                        var result = Sse2.AddSaturate(value.Item1, value.Item2);
+                        sbyteTable.SetOutArray(result);
+                    }
+
+                    for (int i = 0; i < testsCount; i++)
+                    {
+                        (Vector128<byte>, Vector128<byte>, Vector128<byte>) value = byteTable[i];
+                        var result = Sse2.AddSaturate(value.Item1, value.Item2);
+                        byteTable.SetOutArray(result);
+                    }
+
+                    CheckMethod<short> checkInt16 = (short x, short y, short z, ref short a) =>
+                    {
+                        int value = x + y;
+                        a = (short) (value >= short.MaxValue ? short.MaxValue : value <= short.MinValue ? short.MinValue : value);
+                        return a == z;
+                    };
+
+                    if (!shortTable.CheckResult(checkInt16))
+                    {
+                        PrintError(shortTable, methodUnderTestName, "(short x, short y, short z, ref short a) => (a = (short)(x & y)) == z", checkInt16);
+                        testResult = Fail;
+                    }
+
+                    CheckMethod<ushort> checkUInt16 = (ushort x, ushort y, ushort z, ref ushort a) =>
+                    {
+                        int value = x + y;
+                        a = (ushort)(value >= ushort.MaxValue ? ushort.MaxValue : value);
+                        return a == z;
+                    };
+
+                    if (!ushortTable.CheckResult(checkUInt16))
+                    {
+                        PrintError(ushortTable, methodUnderTestName, "(ushort x, ushort y, ushort z, ref ushort a) => (a = (ushort)(x & y)) == z", checkUInt16);
+                        testResult = Fail;
+                    }
+
+                    CheckMethod<sbyte> checkSByte = (sbyte x, sbyte y, sbyte z, ref sbyte a) =>
+                    {
+                        int value = x + y;
+                        a = (sbyte)(value >= sbyte.MaxValue ? sbyte.MaxValue : value <= sbyte.MinValue ? sbyte.MinValue : value);
+                        return a == z;
+                    };
+
+                    if (!sbyteTable.CheckResult(checkSByte))
+                    {
+                        PrintError(sbyteTable, methodUnderTestName, "(sbyte x, sbyte y, sbyte z, ref sbyte a) => (a = (sbyte)(x & y)) == z", checkSByte);
```

Comment:
How is the message used here, it is non-obvious from the surrounding code.

Reply:
I have standardized on error messages in `PrintError` functions, they all use this same header defined in `TestTableSse2.cs` line 1271:
```C#
        private static void PrintErrorHeaderTu<T>(string functionName, string testFuncString) where T : struct
        {
            Console.WriteLine($"{typeof(Sse2)}.{functionName} failed on {typeof(T)}:");
            Console.WriteLine($"Test function: {testFuncString}");
            Console.WriteLine($"{ typeof(Sse2)}.{functionName} test tuples:");
        }
```